other/install_2-1_mysql.sql

```diff
@@ -794,6 +795,7 @@ CREATE TABLE {$db_prefix}messages (
   PRIMARY KEY (id_msg),
   UNIQUE idx_id_board (id_board, id_msg),
   UNIQUE idx_id_member (id_member, id_msg),
+  UNIQUE idx_id_topic (id_topic, id_msg),
```

Comment:
the index INDEX idx_current_topic (id_topic, id_msg, id_member, approved),
already exists so this index is a duplicate (same field in the same order).

Reply:
Is there no need to enforce the UNIQUE constraint?   

This exists in 2.0, so it exists in upgraded forums, which is why I added it back.

Reply:
PK + any field = **not necessarily** unique.  Having one field a pk doesn't make any combo unique.  I don't think I follow your logic there.  

2.0 has these 3 UNIQUE constraints.  

I'm not sure I see a reason to remove any of them.  If we DO want to remove them, we must do it across installs & upgrades, not just one or the other.

Reply:
Show me a case where pk field is unique and pk field + any col is not unique.

Reply:
I see your point...

Should none of the 3 indexes be unique?

The unique was removed from the install, but we would need to take further steps to rebuild it to make upgraded DBs align.

Reply:
I'm realy sure that i'm the person how removed the unique index from installer.
I "see" no issue that the installer is different as the upgrade version,
on the other side some of the stuff could create issues on upgrade progress side (the droping take no time but creation of one can take long time...)

so the change of unique to a normal index can bring up many issue with no benefit.

Reply:
OK, so I'll leave it as-is, where both sets are UNIQUE and consistent.

Reply:
My understanding was that you revert the change of "UNIQUE idx_id_topic (id_topic, id_msg),"
and keep only the both other unique index.

Reply:
Can't do that to the upgrader, as you pointed out above.

Two options....     3 unique indexes (as above), or, live with the inconsistency.  

Which would you prefer?

Reply:
well i describe in the upper comment that is a bad idea to build up a index in the upgrade process,
but to drop doesn't take much time.
So i would take option three: drop the index in the upgrade process, too.

Reply:
Drop it from both the installer and upgrader?

Is the index not needed?

Reply:
The index(UNIQUE idx_id_topic (id_topic, id_msg)) hat possible two function:
- Check if the combination of cols is unique, but we came already to the result that the pk field by his own is already unique -> for this case the index is useless
- Make the access two the col id_topic, id_msg faster, but we got already the index; INDEX idx_current_topic (id_topic, id_msg, id_member, approved) -> for this case already a better index exists

we came to the result both cases where the index could be needed already catch up -> no need for this index.

Reply:
Ok.  Will do.